integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala

```diff
@@ -184,6 +189,161 @@ case class AlterTableCompaction(alterTableModel: AlterTableModel) extends Runnab
+      .getOrElse(sparkSession.catalog.currentDatabase)
+    val splitInfo = alterTableSplitPartitionModel.splitInfo
+    val partitionId = Integer.parseInt(alterTableSplitPartitionModel.partitionId)
+    val timestampFormatter = new SimpleDateFormat(CarbonProperties.getInstance
+      .getProperty(CarbonCommonConstants.CARBON_TIMESTAMP_FORMAT,
+        CarbonCommonConstants.CARBON_TIMESTAMP_DEFAULT_FORMAT))
+    val dateFormatter = new SimpleDateFormat(CarbonProperties.getInstance
+      .getProperty(CarbonCommonConstants.CARBON_DATE_FORMAT,
+        CarbonCommonConstants.CARBON_DATE_DEFAULT_FORMAT))
+
+    val locksToBeAcquired = List(LockUsage.METADATA_LOCK,
+      LockUsage.COMPACTION_LOCK,
+      LockUsage.DELETE_SEGMENT_LOCK,
+      LockUsage.DROP_TABLE_LOCK,
+      LockUsage.CLEAN_FILES_LOCK,
+      LockUsage.ALTER_PARTITION_LOCK)
+    var locks = List.empty[ICarbonLock]
+    try {
+      locks = AlterTableUtil.validateTableAndAcquireLock(dbName, tableName,
+        locksToBeAcquired)(sparkSession)
+      val carbonMetastore = CarbonEnv.getInstance(sparkSession).carbonMetastore
+      val relation = carbonMetastore.lookupRelation(Option(dbName), tableName)(sparkSession)
+        .asInstanceOf[CarbonRelation]
+      val carbonTableIdentifier = relation.tableMeta.carbonTableIdentifier
+      val storePath = relation.tableMeta.storePath
+      if (relation == null) {
+        sys.error(s"Table $dbName.$tableName does not exist")
+      }
+      carbonMetastore.checkSchemasModifiedTimeAndReloadTables(storePath)
+      if (null == CarbonMetadata.getInstance.getCarbonTable(dbName + "_" + tableName)) {
+        LOGGER.error(s"Alter table failed. table not found: $dbName.$tableName")
+        sys.error(s"Alter table failed. table not found: $dbName.$tableName")
+      }
+      val carbonLoadModel = new CarbonLoadModel()
+
+      val table = relation.tableMeta.carbonTable
+      val partitionInfo = table.getPartitionInfo(tableName)
+      val partitionIdList = partitionInfo.getPartitionIds.asScala
+      // keep a copy of partitionIdList before update partitionInfo.
+      // will be used in partition data scan
+      val oldPartitionIdList: ArrayBuffer[Int] = new ArrayBuffer[Int]()
+      for (i: Integer <- partitionIdList) {
+        oldPartitionIdList.append(i)
+      }
+
+      if (partitionInfo == null) {
+        sys.error(s"Table $tableName is not a partition table.")
+      }
+      if (partitionInfo.getPartitionType == PartitionType.HASH) {
+        sys.error(s"Hash partition table cannot be added or split!")
+      }
+      /**
+       * verify the add/split information and update the partitionInfo:
+       *  1. update rangeInfo/listInfo
+       *  2. update partitionIds
+       */
+      val columnDataType = partitionInfo.getColumnSchemaList.get(0).getDataType
+      val index = partitionIdList.indexOf(partitionId)
+      if (partitionInfo.getPartitionType == PartitionType.RANGE) {
```

Comment:
try to create more private functions and use it in `run`, thus `run` is shorter, better for readability